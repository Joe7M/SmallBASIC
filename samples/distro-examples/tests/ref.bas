rem reference variables not supported
a = "dog"
b = @a
a = "cat"
print "cat=", a
print "dog=", b

rem complex pseudo class method references
func C
  func f(j)
    return j
  end
  local r = {}
  r.f = @f
  return r
end
func Q
  local r = {}
  r.c = C()
  return r
end
c.m = Q()
j = [1,2,3]
for n in c.m.c.f(j)
  print n
next n
