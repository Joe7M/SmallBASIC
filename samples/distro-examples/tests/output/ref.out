cat=	cat
dog=	dog
10
20
